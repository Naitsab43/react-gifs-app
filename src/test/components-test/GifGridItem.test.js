const { shallow } = require("enzyme")
const { GifGridItem } = require("../../components/GifGridItem");
import React from "react";
import "@testing-library/jest-dom";


describe('Pruebas a <GifGridItem />', () => {
  
  // Global 

  const title = "Soy un titulo";
  const url = "https://localhost/naruto.jpg";

  const wrapper = shallow(<GifGridItem title={title} url={url}/>);

  test('Debe de mostrar el componente correctamente', () => {
    
    
    expect(wrapper).toMatchSnapshot();

  })


  test('Debe de tener un parrafo con el title', () => {
    
    const p = wrapper.find("p");

    expect(p.text().trim()).toBe(title);


  });

  test('Debe de tener una imagen igual al src y al alt', () => {
    
    const img = wrapper.find("img");

    expect(img.prop("src")).toBe(url);

    expect(img.prop("alt")).toBe(title);


  });
  
  test('Debe tener la clase animate__fadeIn', () => {
    
    const div = wrapper.find("div");

    expect(div.hasClass("animate__fadeIn")).toBe(true);

  })
  


})
