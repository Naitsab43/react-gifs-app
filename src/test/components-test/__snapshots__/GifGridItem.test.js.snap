// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas a <GifGridItem /> Debe de mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <p
    className="card__title"
  >
    Soy un titulo
  </p>
  <img
    alt="Soy un titulo"
    className="card__img"
    src="https://localhost/naruto.jpg"
  />
</div>
`;
